OpenRA.Mods.Common/Traits/World/WeatherOverlay.cs

```diff
@@ -0,0 +1,391 @@
+							Game.CosmeticRandom.Next(info.SwingSpeed[0], info.SwingSpeed[1] + 1) / 1000f,
+							tempSwingDirection,
+							Game.CosmeticRandom.Next(info.SwingAmplitude[0], info.SwingAmplitude[1]) / 10f,
+							tempColor,
+							tempColorTail));
+				}
+			}
+		}
+
+		void ParticlesCountLogic(WorldRenderer wr)
+		{
+			// Logic
+			if (particleCountFader == ParticleCountFaderType.FadeIn && ParticleList.Count >= maxParticleCount)
+				particleCountFader = ParticleCountFaderType.Hold;
+			else if (particleCountFader == ParticleCountFaderType.FadeOut && ParticleList.Count == 0)
+				particleCountFader = ParticleCountFaderType.Hold;
+
+			// Do the fade functions
+			if (particleCountFader == ParticleCountFaderType.FadeIn)
+				FadeInCountParticle(wr);
+			else if (particleCountFader == ParticleCountFaderType.FadeOut)
+				FadeOutCountDoPa(wr);
+		}
+
+		void FadeInCountParticle(WorldRenderer wr)
+		{
+			SpawnParticles(1, 0, 100);
+
+			// Remove Particles, which are getting replaced from the top to the bottom by the "EdgeCheckReplace",
+			// when scrolling down, as long as the FadeIn is not completed,
+			// to avoid having particles at the top and bottom, but not in the middle of the screen.
+			for (int i = 0; i < ParticleList.Count; i++)
+			{
+				if (ParticleList[i].PosY < 0)
+					ParticleList.RemoveAt(i);
+			}
+
+			// Add Particles when the weather is fading in and scrolling up, to fill areas above
+			if (antiScrollPrevTopLeft.Y > wr.Viewport.TopLeft.Y)
+			{
+				// Get delta Y and limit to the max value
+				var tempRangeY = antiScrollPrevTopLeft.Y - wr.Viewport.TopLeft.Y;
+				var tempParticleCount = CalculateParticleCount(Game.Renderer.Resolution.Width, (int)tempRangeY);
+				if (ParticleList.Count + tempParticleCount > maxParticleCount)
+					tempParticleCount = maxParticleCount - ParticleList.Count;
+				SpawnParticles(tempParticleCount, (int)tempRangeY, 50);
+			}
+		}
+
+		void FadeOutCountDoPa(WorldRenderer wr)
+		{
+			for (int i = 0; i < ParticleList.Count; i++)
+			{
+				if (ParticleList[i].PosY > (Game.Renderer.Resolution.Height - ParticleList[i].Gravity))
+					ParticleList.RemoveAt(i);
+			}
+		}
+
+		Particle XAxisSwing(Particle tempParticle)
+		{
+			// Left direction turn
+			if (tempParticle.SwingOffset < (-1 * tempParticle.SwingAmplitude))
+				tempParticle.SwingDirection = !tempParticle.SwingDirection;
+
+			// Right direction turn
+			else if (tempParticle.SwingOffset > tempParticle.SwingAmplitude)
+				tempParticle.SwingDirection = !tempParticle.SwingDirection;
+
+			// Performe the X-Axis-Swing
+			if (tempParticle.SwingDirection)
+				tempParticle.SwingOffset += tempParticle.SwingSpeed;
+			else
+				tempParticle.SwingOffset -= tempParticle.SwingSpeed;
+
+			return tempParticle;
+		}
+
+		public void Tick(Actor self)
+		{
+			windTickCountdown--;
+		}
+
+		Particle WindLogic(Particle tempParticle)
+		{
+			if (info.ChangingWind)
+			{
+				if (windTickCountdown <= 0)
```

Comment:
if you change the if to !info.ChangingWind and switch the if else statement you can change the above to else if (windTickCountdown <= 0).